build.py

```diff
@@ -0,0 +1,188 @@
+
+sys.path.append('./pylibs')
+try:
+    import watchdog
+except ModuleNotFoundError:
+    pass  # Not mandatory
+
+
+
+EXTENSION_ZIP_NAME = 'NopeCHA.zip'
+BASE_MANIFEST = Path('manifest.base.json')
+EXPORT_PATH = Path('__export__')
+VERSIONS_PATH = Path('version')
+
+
+watchdog_help = (f"""
+Keeps the program running and makes use of the watchdog library to listen for changes. 
+When changes exist, the corresponding file is updated in {EXPORT_PATH} for the extension to update itself and use.
+""").strip()
+
+WATCHDOG_NOT_FOUND = f"""
+    watchdog was not found. Execute the command below if you want to use the -w/--watch feature:
+    python -m pip install -t pylibs watchdog
+""".strip()
+
+try:
+    watchdog
+except NameError:
+    watchdog_help += f"""
+Warning:
+{WATCHDOG_NOT_FOUND}
+"""
+
+parser = argparse.ArgumentParser(
+    description='Dev/deploy build extension for firefox and chrome',
+    formatter_class=argparse.RawTextHelpFormatter
+)
+
+parser.add_argument('-d', '--deploy', action="store_true", help="Compresses files and creates zip file to submit for evaluation or test in the browser")
+parser.add_argument('-w', '--watch', action="store_true", help=watchdog_help)
+
+program_args = parser.parse_args()
+
+try:
+    watchdog
+except NameError:
+    if program_args.watch:
+        parser.print_help()
+        printe(f"\nERROR: \n {WATCHDOG_NOT_FOUND}\n")
+        exit(2)
+
+@contextlib.contextmanager
+def in_dir(new_path):
+    oldcwd = os.getcwd()
+    try:
+        os.chdir(new_path)
+        yield
+    finally:
+        os.chdir(oldcwd)
+
+dir_path = Path(__file__).absolute().parent
+
+with in_dir(dir_path):
+
+    def deploy():
+        here_path = Path('.')
+
+        # This is slow but there should be very few files, so it should be OK
+        files_to_include = list(itertools.chain(
+            here_path.glob('*.html'),
+            here_path.glob('*.css'),
+            here_path.glob('*.js'),
+            (here_path / 'icon').iterdir(),
+        ))
+
+        versions = filter(lambda p: p.is_dir(), VERSIONS_PATH.iterdir())
+        # This could be run just once per program run but it would bring unnecessary complexity to the code below
+        shutil.rmtree(EXPORT_PATH, True)
+
+        for version in versions:
+            export_directory = EXPORT_PATH / version
+            export_directory.mkdir(parents=True, exist_ok=True)
+
+            if program_args.deploy:
+                # Takes quite a while to make new zips. So better do it for final tests or when deploying
+                zip_deploy = ZipFile(export_directory / EXTENSION_ZIP_NAME, 'w', ZIP_DEFLATED, compresslevel=9)
+            else:
+                zip_deploy = contextlib.nullcontext()
+
+            with zip_deploy as zip:
+
+                printe('-' * 80)
+                printe('packaging version', version)
+
+                with BASE_MANIFEST.open('r') as f:
+                    extension_manifest = json.load(f)
```

Comment:
just personal preference, feel free to ignore
```suggestion
                extension_manifest = json.loads(BASE_MANIFEST.read_text())
```

Reply:
It does look good to me, though. I'm buying it ^^